metakb/service_schema.py

```diff
@@ -0,0 +1,259 @@
+"""Define schemas for MetaKB service"""
+from typing import List, Optional, Union, Dict, Any, Type
+
+from ga4gh.vrsatile.pydantic.vrs_models import CURIE
+from ga4gh.vrsatile.pydantic.vrsatile_models import GeneDescriptor, \
+    VariationDescriptor, TherapeuticDescriptor, DiseaseDescriptor
+from pydantic import BaseModel
+from pydantic.types import StrictBool, StrictStr
+
+
+from metakb.schemas import Direction, VariationGermlinePathogenicityProposition, \
+    VariationGermlinePathogenicityStatement, \
+    VariationNeoplasmTherapeuticResponseProposition, \
+    VariationNeoplasmTherapeuticResponseStatement, Method, Document, VariationOrigin
+from metakb.version import __version__, LAST_UPDATED
+
+
+class Response(BaseModel):
+    """Define the Response Model."""
+
+    statements: List[Union[VariationGermlinePathogenicityStatement,
+                           VariationNeoplasmTherapeuticResponseStatement]]
+    target_propositions: List[Union[VariationGermlinePathogenicityProposition,
+                                    VariationNeoplasmTherapeuticResponseProposition]]
+    variation_descriptors: List[VariationDescriptor]
+    gene_descriptors: List[GeneDescriptor]
+    therapeutic_descriptors: Optional[List[TherapeuticDescriptor]]
+    disease_descriptors: List[DiseaseDescriptor]
+    methods: List[Method]
+    documents: List[Document]
+
+
+class NestedStatementResponse(BaseModel):
+    """Define Statement Response for Search Endpoint."""
+
+    id: CURIE
+    type: StrictStr
+    description: StrictStr
+    direction: Optional[Direction]
+    evidence_level: CURIE
+    variation_origin: Optional[VariationOrigin]
+    target_proposition: List[Union[VariationGermlinePathogenicityProposition,
+                                   VariationNeoplasmTherapeuticResponseProposition]]
+    variation_descriptor: VariationDescriptor
+    therapeutic_descriptor: Optional[TherapeuticDescriptor]
+    disease_descriptor: DiseaseDescriptor
+    method: Method
+    documents: List[Document]
+
+    class Config:
+        """Configure examples."""
+
+        @staticmethod
+        def schema_extra(schema: Dict[str, Any],
+                         model: Type['NestedStatementResponse']) -> None:
+            """Configure OpenAPI schema"""
+            if 'title' in schema.keys():
+                schema.pop('title', None)
+            for prop in schema.get('properties', {}).values():
+                prop.pop('title', None)
+            # TODO: Update
```

Comment:
? is this pending an Alex contribution

there are also some other TODO: updates on this page as well fwiw

Reply:
Oh, I forgot to note that the service_schema won't be addressed til #193 . This was just a copy/paste into a new file.